homeassistant/components/sensor/dht.py

```diff
@@ -39,9 +39,8 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         "DHT22": Adafruit_DHT.DHT22,
         "AM2302": Adafruit_DHT.AM2302
     }
-    sensor = available_sensors[config['sensor']]
-
-    pin = config['pin']
+    sensor = available_sensors.get(config.get('sensor'))
+    pin = config.get('pin')
```

Comment:
1. 'pin' should be a constant at this point
2. If pin is not set in the config - how is the plat form supposed to work?